test/parallel/test-stream-pipeline-duplex.js

```diff
@@ -1,21 +1,39 @@
 'use strict';
 
 const common = require('../common');
-const { pipeline, Duplex, PassThrough } = require('stream');
+const { pipeline, Duplex, PassThrough, Readable } = require('stream');
 const assert = require('assert');
 
-const remote = new PassThrough();
-const local = new Duplex({
-  read() {},
-  write(chunk, enc, callback) {
-    callback();
-  }
-});
+{
+  // Call pipeline error function with ERR_STREAM_PREMATURE_CLOSE error if dst close before src
+  const remote = new PassThrough();
+  const local = new Duplex({
+    read() {
+    },
+    write(chunk, enc, callback) {
+      callback();
+    },
+  });
 
-pipeline(remote, local, remote, common.mustCall((err) => {
-  assert.strictEqual(err.code, 'ERR_STREAM_PREMATURE_CLOSE');
-}));
+  pipeline(remote, local, remote, common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ERR_STREAM_PREMATURE_CLOSE');
+  }));
 
-setImmediate(() => {
-  remote.end();
-});
+  setImmediate(() => {
+    remote.end();
+  });
+}
+
+
+{
+  // Must call when last stream is Duplex from function
+  pipeline(
+    Readable.from(['a', 'b', 'c', 'd']),
+    Duplex.from(async function* stopAfterFirst(stream) {
+      for await (const chunk of stream) {
+        yield chunk;
+      }
+    }),
+    common.mustSucceed(),
+  ).on('error', common.mustNotCall());
```

Comment:
I don't understand this test.

Reply:
the pipeline callback is not called

Reply:
`stopAfterFirst`... but it doesn't stop after first?

Reply:
yes, from other debugging, removed, see latest commit

Reply:
?

Reply:
the latest commit does not have the `stopAfterFirst` function name